chainer_setup_build.py

```diff
@@ -193,125 +159,83 @@ def make_extensions(options, compiler):
-
-
 def parse_args():
-    global _arg_options
-    _arg_options['profile'] = '--cupy-profile' in sys.argv
-    if _arg_options['profile']:
+    arg_options = dict()
+    arg_options['profile'] = '--cupy-profile' in sys.argv
+    if arg_options['profile']:
         sys.argv.remove('--cupy-profile')
 
     cupy_coverage = '--cupy-coverage' in sys.argv
     if cupy_coverage:
         sys.argv.remove('--cupy-coverage')
-    _arg_options['linetrace'] = cupy_coverage
-    _arg_options['annotate'] = cupy_coverage
+    arg_options['linetrace'] = cupy_coverage
+    arg_options['annotate'] = cupy_coverage
 
-    _arg_options['no_cuda'] = '--cupy-no-cuda' in sys.argv
-    if _arg_options['no_cuda']:
+    arg_options['no_cuda'] = '--cupy-no-cuda' in sys.argv
+    if arg_options['no_cuda']:
         sys.argv.remove('--cupy-no-cuda')
     if check_readthedocs_environment():
-        _arg_options['no_cuda'] = True
-
-
-def get_cython_pkg():
-    try:
-        return pkg_resources.get_distribution('cython')
-    except pkg_resources.DistributionNotFound:
-        return None
+        arg_options['no_cuda'] = True
+    return arg_options
 
 
-def run_command(cmd):
+def check_cython_version():
     try:
-        subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-    except subprocess.CalledProcessError as e:
-        msg = '''Command %r failed:
-
-  command: %s
-  return code: %d
-  output:
-
-%s''' % (cmd[0], ' '.join(e.cmd), e.returncode, e.output)
-        raise distutils.errors.DistutilsExecError(msg)
-
+        import Cython
+        cython_version = pkg_resources.parse_version(Cython.__version__)
+        return cython_version >= required_cython_version
+    except ImportError:
+        return False
 
-def cythonize(
-        extensions, force=False, annotate=False, compiler_directives=None):
-    cython_location = get_cython_pkg().location
-    cython_path = path.join(cython_location, 'cython.py')
-    print("cython path:%s" % cython_location)
-    cmd = [sys.executable, cython_path]
-    run_command(cmd + ['--version'])
 
-    cmd.extend(['--fast-fail', '--verbose', '--cplus'])
-    if compiler_directives is not None:
-        for i in compiler_directives.items():
-            cmd.append('--directive')
-            cmd.append('%s=%s' % i)
+def cythonize(extensions, arg_options):
+    import Cython.Build
 
-    for ext in extensions:
-        run_command(cmd + ext.sources)
+    directive_keys = ('linetrace', 'profile')
+    directives = {key: arg_options[key] for key in directive_keys}
 
+    cythonize_option_keys = ('annotate',)
+    cythonize_options = {key: arg_options[key]
+                         for key in cythonize_option_keys}
 
-def to_cpp_extensions(extensions):
-    ret = []
-    for x in extensions:
-        ext = copy.copy(x)
-        ext.sources = [path.splitext(f)[0] + ".cpp" for f in x.sources]
-        ret.append(ext)
-    return ret
+    return Cython.Build.cythonize(
+        extensions, language="c++", verbose=True,
```

Comment:
`language` argument is deprecated. https://github.com/cython/cython/blob/f93e8de11de0f6e7236d27fad1806104f354ad77/Cython/Build/Dependencies.py#L638